willa/tind/fetch.py

```diff
@@ -62,3 +68,86 @@ def fetch_file(file_url: str, output_dir: str = '') -> str:
         raise RecordNotFoundError('Referenced file could not be downloaded.')
 
     return saved_to
+
+def fetch_file_metadata(record: str) -> Record:
+    """Fetch file metadata for a given Tind record.
+    :raises AuthorizationError: When the TIND API key is invalid.
+    :raises Exception: for any response other then 200. 
+    :returns a json list of metadata for a given Tind record
+    """ 
+
+    status, files = tind_get(f"record/{record}/files")
+
+    if status != 200 and status < 500:
+        j = json.loads(files)
+        reason = j['reason']
+        raise TINDError(f"Status: {status} Message: {reason}.")
+    
+    return json.loads(files)
+
+def fetch_ids_search(search: str):
+    status, rec_ids = tind_get(f"search", {'p': search})
+
+    if status >= 400:
+        j = json.loads(rec_ids)
+        reason = j['reason']
+        raise HTTPError(f"Status: {status}, Message: {reason}")
+
+    j = json.loads(rec_ids) 
+    return j['hits']
+
+def fetch_marc_by_ids(ids: list):
+    """Fetch Tind marc from a list of Tind record ids
+    :returns a list of PYMARC records 
+    """ 
+    records = []
+    for item in ids:
+      m = fetch_metadata(item)
+      records.append(m)
+
+    return records
+
+def fetch_search_metadata(search: str):
+    ids = fetch_ids_search(search)
+
+    return fetch_marc_by_ids(ids)
+
+def search_request(search: str, search_id=None ):
+    if search_id:
+      status, response = tind_get('search', {'format': 'xml', 'p': search, 'search_id': search_id})
+    else:
+      status, response = tind_get('search', {'format': 'xml', 'p': search})
+    
+    return response
+
+def retrieve_xml_search_id(response):
+  E.register_namespace('', "http://www.loc.gov/MARC21/slim")
+  xml = E.fromstring(response)
+  search_id = xml.find('search_id').text
+
+  return xml, search_id
```

Comment:
Make the indentation match the rest of the file.  This seems to be a private method, so add a _ to the beginning.  This is somewhat like marking a Python method 'private' (though it doesn't actually prevent external callers).

```suggestion

def _retrieve_xml_search_id(response):
    E.register_namespace('', "http://www.loc.gov/MARC21/slim")
    xml = E.fromstring(response)
    search_id = xml.find('search_id').text

    return xml, search_id
```